src/basic/path-lookup.h

```diff
@@ -52,6 +52,9 @@ struct LookupPaths {
 
         /* A temporary directory when running in test mode, to be nuked */
         char *temporary_dir;
+
+        /* search path -> hashmap of symlinks */
+        Hashmap *search_path_link_cache;
```

Comment:
hmm, PathLookup is a strange place for this I think, since it is used by the non-install stuff too, to search for units in PID 1 when loading them. 

Moreover, we path LookupPaths around in "const" fashion everywhere in install.c, i.e. the assumption is that the data is constant, but it actually isn't. of course, "const" is not supposed to cover stuff pointed to from a struct, but conceptually it's still weird to modify something that belongs, is owned by a const object.

I'd really just pass this around separately whereever appropruate in install.c. Yes the functions already have a ton of parameters, but addng one more is till better than adding this to LookupPaths...

(we might want to introduce an `InstallContext` object or so that we can share between multiple operations, but that's probably something for a different PR, given it requires some non-trivial refactoring)

Reply:
the comment here is wrong (or just confusing)? it's a hashtable from `symlink path → destination path` no?